src/Button/index.ts

```diff
@@ -0,0 +1,9 @@
+export {Button as default} from './Button'
+export {Button} from './Button'
```

Comment:
Should the props types be exported here too?

Reply:
There’s no right answer; it depends on the ergonomics we want to provide. It’s common for barrels to re-export all exports (which would include type exports), e.g. from the [TypeScript Deep Dive: Barrel]((https://basarat.gitbook.io/typescript/main-1/barrel)) page:

```ts
// demo/index.ts
export * from './foo'; // re-export all of its exports
export * from './bar'; // re-export all of its exports
export * from './baz'; // re-export all of its exports
```

Reply:
> I think this specific syntax is incompatible with default exports

The answer to this isn’t clearer to me after reading the [MDN `export` page](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export) and the [ECMA-262 spec](https://tc39.es/ecma262/#sec-exports).

If you `export *` from two files which each have a `default` export, which won is bound to the barrel’s `default` export?

Reply:
I tested this in a [sandbox](https://codesandbox.io/s/nostalgic-feynman-6ndwb?file=/src/index.ts). I get two errors that `Module '"/sandbox/src/module-aggregate"' has no default export.` and `'export *' does not re-export a default.`, but _it still works_. 😱 That is, the first re-exported function appears to be bound to the aggregate (barrel)’s `default`, contra the error messages.

Reply:
Setting `export *` aside, we can go ahead and export the named prop types explicitly. Added this via https://github.com/primer/components/pull/1017/commits/e3aed73876ed0e2f373aed9898ba09f381cbc68b.